### ISSUE TITLE
[FR] Toolbox / Insert markdown file from vault / link Markdown Blocks or Headings and not full file

### ISSUE THREAD

**Issue by stefnestor** [zsviczian/obsidian-excalidraw-plugin#745]
👋! Will you consider adding the ability for the toolbox feature "insert markdown file from vault" to connect to `FILENAME^ID` (+ `FILENAME#HEADER`) directly? (Related: [Obsidian's Link to Blocks](https://help.obsidian.md/How+to/Link+to+blocks), [Obsidian's Internal Link to Headings](https://help.obsidian.md/How+to/Internal+link))

Currently only links to `FILENAME.md` and then you need to manually edit the backend Markdown, e.g. :

1. Create a `FILENAME.md`
2. In current/another file, generate a link`[[FILENAME^ID]]`
3. In new Excalidraw, click to "insert markdown file from vault" for `FILENAME.md` twice
4. Click to "Open as Markdown" and replace on insertion from `[[FILENAME.md]]` to `[[FILENAME^ID]]`

Results in 

![image](https://user-images.githubusercontent.com/26751266/185178392-6c15c5f2-cc5f-4c80-87bd-8518bc41635a.png)

**Comment by zsviczian** [comment]
absolutely. This has been bugging me as well. This is a good nudge to make it happen.

**Comment by you-n-g** [comment]
After we have `insert ANY file` feature,  I think the following workflow would be better for the tool

1. Editing on  `filename.excalidraw.md`
2. Invote the tool to Insert markdown and enter a heading name (or generate an anchor ID)
3. Append the heading into  `filename.excalidraw.text.md`  and insert the section into current `filename.excalidraw.md` via `insert ANY file`.

This would provide similar but amazing feature like Obsidian canvas  in excalidraw.

Looking forward to the new feature!
